lib/identity_cache/configuration_dsl.rb

```diff
@@ -233,7 +237,14 @@ def #{options[:cached_ids_name]}
           end
 
           def #{options[:cached_accessor_name]}
-            if IdentityCache.should_use_cache? && !#{association}.loaded?
+            association_klass = association(:#{association}).klass
+            should_use_cache = if association_klass.respond_to?(:should_use_cache?)
```

Comment:
Why is this `respond_to?` check necessary?  This is for cache_has_many with `embed: :ids` where those ids will be loaded using the associated model, thus that model must already `include IdentityCache`